### PR TITLE
fix(dropdown): Swallows space keys after it's been opened

### LINKED ISSUE #604: Dropdown (v2): Swallows space keys after it's been opened
## Detailed description

> Describe in detail the issue you're having.

I have a dropdown with an item template which adds a search box in the items list. The issue is that when I type something in the search box it gets through the `Dropdown.onKeyDown()` method first which swallows the spaces.

image

> Is this a feature request (new component, new icon), a bug, or a general issue?

Bug

> Is this issue related to a specific component?

Dropdown

> What did you expect to happen? What happened instead? What would you like to see changed?

I expect the component to handle spaces only when it needs to open the menu and leave them to propagate otherwise.

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

2.45.0

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

IBM ADDI

### PR DESCRIPTION
**Pull request by youda97**
closes #604

### CODE REVIEW

File: src/dropdown/list/dropdown-list.component.ts [hunk 1]
```diff
@@ -405,10 +406,12 @@ export class DropdownList implements AbstractDropdownView, AfterViewInit, OnDest
 	doKeyDown(event: KeyboardEvent, item: ListItem) {
 		// "Spacebar", "Down", and "Up" are IE specific values
 		if (event.key === "Enter" || event.key === " " || event.key === "Spacebar") {
-			event.preventDefault();
-			if (event.key === "Enter") {
-				this.doClick(event, item);
-			}
+				if (this.listElementList.some(option => option.nativeElement === event.target)) {
+					event.preventDefault();
```
Comment (cal-smith) [inline src/dropdown/list/dropdown-list.component.ts]: Did you check this with an `input` as a custom template? That's where the original issue ran into problems ...

File: src/dropdown/list/dropdown-list.component.ts [hunk 2]
```diff
@@ -67,7 +65,10 @@ import { Observable, isObservable, Subscription } from "rxjs";
 					'bx--list-box__menu-item--active': item.selected,
 					disabled: item.disabled
 				}">
-				<div class="bx--list-box__menu-item__option">
+				<div
+					#listItem
+					tabindex="0"
```
Comment (cal-smith) [inline src/dropdown/list/dropdown-list.component.ts]: Should this be `-1`? AFAIK we don't want users tabbing through the list

File: src/dropdown/list/dropdown-list.component.ts (hunk 1 above)
Comment (youda97) [reply src/dropdown/list/dropdown-list.component.ts]: Yup I tested it with the search and it was working

Comment (cal-smith) [reply src/dropdown/list/dropdown-list.component.ts]: Awesome, we should be good to go then

**Pull request merged**